packages/ERTP/src/payment.js

```diff
@@ -0,0 +1,24 @@
+/* global makeKind */
+
+// @ts-check
+
+import { Far } from '@agoric/marshal';
+import { makeFarName, ERTPKind } from './interfaces';
+
+export const makePaymentMaker = (allegedName, brand) => {
+  const paymentVOMaker = state => {
+    return {
+      init: b => (state.brand = b),
+      self: Far(makeFarName(allegedName, ERTPKind.PAYMENT), {
+        getAllegedBrand: () => state.brand,
+      }),
```

Comment:
Not sure how well the `Far(makeFarName(...` stuff is going to get along with the virtual object machinery.  The construct is certainly ugly and very hard to parse visually.  I would have written this as just: 
```
  self: {
    getAllegedBrand: () => state.brand,
  }
```
But if this object is going to be used as a presence we need to get that `Far()` wrapper on there somehow.

The return value from `paymentVOMaker` is going to get hardened.  I don't know how `harden` deals with objects with properties that are already hardened.  I *think* that's OK, but this is one of those mysterious edge cases that I always have to either ask @erights about or just try and see what happens.

Reply:
`harden` is idempotent. That part is fine.

Reply:
I was expecting the `Far...` to be in the `makeKind` invocation.

I don't like needing the `self:` section. There was a slightly different pattern that didn't require that, but I couldn't lay my hands on that issue yet.

Reply:
As @dtribble said, `harden()` is idempotent. However `Far` is not: it only accepts *unhardened* objects (it mutates them before hardening them).

I have a branch in which `makeKind` asserts that the maker it was given is applying `Far` to the `self` that it returns, under the theory that the maker should be doing `harden` to the `{ init, self }` pair that it returns ("don't let objects out of your sight without hardening them first"), and `Far()` cannot be applied to objects that are already hardened, therefore the maker must apply the `Far` before it hardens `{ init, self }`.

(the assertion on that branch isn't very helpful, however, because if you make the most common mistake and don't harden the return value, it emits a particularly weird error message instead of something that would guide you towards the light)

It would certainly be more ergonomic to tell `makeKind` the interface name, and then have `makeKind` apply the `Far()` itself, but that would require that the maker *not* harden its return value, which feels counter to the habit we're trying to teach developers.

Reply:
I think that integration is the right thing.  It should happen within the same module. The `init/self` is the squishy underbelly of constructing useful remote and persistent objects. They don't need to be hardened, given that they should never escape directly.

Reply:
> They don't need to be hardened, given that they should never escape directly.

@dtribble ,
I completely disagree with this sentiment. Any object that has another object as a client has "escaped". An unfrozen object cannot feasibly be defensively consistent. To build robust systems, all component abstractions should by default be defensively consistent. Any object that isn't defensively consistent must be carefully encapsulated within a defensively consistent subsystem, and must not escape from that subsystem to the clients of that subsystem.

We have a notational cost for hardening. And we have an adoption cost when explaining `harden`. But we need to weigh these against the costs of having a less robust system.

Unfortunately we won't have the static checking needed to get us to reliably harden anytime soon. But we must still try.

Reply:
(Retracted previous comment. I took "escape" out of context)

Reply:
Oh good. To clarify for the future, I just meant that the internal description of what needs to be built -- `paymentVOMaker`  -- on line 9 gets used on line 18 and the result of that is what gets locked down by the process. The `paymentVOMaker` should never escape.